packages/app/src/elements/settings-tools.ts

```diff
@@ -29,11 +29,15 @@ export class SettingsTools extends StateMixin(LitElement) {
         const file = this._fileInput.files![0];
         const reader = new FileReader();
         reader.onload = async () => {
-            await this._importDialog.show(reader.result as string);
+            await this._importDialog.show(reader.result as string | Uint8Array);
             this._fileInput.value = "";
         };
 
-        reader.readAsText(file);
+        if (file.name.endsWith('.1pux')) {
+            reader.readAsArrayBuffer(file);
+        } else {
+            reader.readAsText(file);
+        }
```

Comment:
Basically because `.1pux` is a zip file, it can't be read as text/string, which requires the type additions above and below.

Reply:
I think the better option would be to defer reading the file at this point and pass around the file object instead. `guessFormat` would then accept a file object for example, and so would `as1Pux` and all the other import functions. This would also allow us to use the file extension and mime type for guessing the format. What do you think?

Reply:
I'm happy to make that change, I just wasn't sure if you'd be fine with changing the API too much.